pennylane/pauli/grouping/optimize_measurements.py

```diff
@@ -34,13 +41,14 @@ def optimize_measurements(observables, coefficients=None, grouping="qwc", colour
     fully-commuting measurement-partitioning approaches respectively.
 
     Args:
-        observables (list[Observable]): a list of Pauli words (Pauli operation instances and Tensor
+        observables (list[Observable]): a list of Pauli words (Pauli operation instances and Hamiltonian
```

Comment:
```suggestion
        observables (list[Observable]): a list of Pauli words (Pauli operation instances and tensor
```
This doesn't work with sums - it needs to a list of Pauli operations and tensors of Pauli operations. Multi-term Hamiltonians can be Pauli sentences, but not Pauli words. 

As an example, `qml.pauli.compute_partition_indices([qml.X(0), qml.Y(1) + qml.Y(0)])` raises an error about expecting a Pauli observable instance, whereas `qml.pauli.compute_partition_indices([qml.X(0), qml.Y(1) @ qml.Y(0)])` works as expected.

Reply:
Very good point, thanks for catching this!

I actually removed the `Tensor` word because the corresponding class was deprecated during this release. I can't think of a better term to use in this case rather than `tensors` with the lowercase letter